fms_extras/models/speculator.py

```diff
@@ -170,3 +170,94 @@ def forward(
             state = self.activation(self.ln[i](state))  # b n d
             out.append(self.head[i](state))  # b n v
         return torch.stack(out, dim=0)  # h b n v
+
+
+def select_inflate_dim(
```

Comment:
the comments detail the mechanics of the operations being performed by this function, but it might help to have a comment with some more general/intuitive sense of the intent of the function (what its purpose is).

Reply:
> if it's only used by generate maybe it should be `__` prefixed?

This is used externally as well, we just have a single example in our repo in speculative_generate that uses it

Reply:
I've changed this to `apply_index_map` to make the intention clearer, and added a higher level description